docs/tutorials/blog.md

```diff
@@ -485,37 +457,31 @@ Update the `app/models/post.server.ts` file:
 ```tsx filename=app/models/post.server.ts lines=[3,9-11]
 import { prisma } from "~/db.server";
 
-export type { Post } from "@prisma/client";
-
 export async function getPosts() {
   return prisma.post.findMany();
 }
 
-export async function getPost(slug: string) {
+export async function getPost(slug?: string) {
```

Comment:
ℹ️ The `slug` parameter should not be made optional here.

If the problem is that `params.slug` is typed as `string|undefined`, you should instead do something along those lines:

```js
export async function loader({ params }: LoaderArgs) {
  if (typeof params.slug === "undefined") {
      throw new Response("Missing slug", 400);
  }

  const post = await getPost(params.slug); // <= from now on TS knows that slug is a `string`

```

Reply:
Got it, I was curious whether I did that right.

Reply:
Rule of thumb is: never change a parameter signature to satisfy TypeScript. 

Use the type that makes sense (here, we do need a slug as a string) and then, make typescript understand that you're calling your function correctly.

Reply:
I took a look at the example right after completing the `getPost` code, it explains about TS showing the error, the next section is about making TS happy and adding the code below. I will remove the `?`. The rest is already explained and the possibility of slug being undefined is handled.

https://github.com/remix-run/remix/blob/493b6a06d483562840ffc5e41311f8c7d5bb5c7c/docs/tutorials/blog.md?plain=1#L546-L556